app/src/main/java/com/infomaniak/mail/views/AttachmentDetailsView.kt

```diff
@@ -0,0 +1,95 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.views
+
+import android.content.Context
+import android.text.format.Formatter
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.annotation.DimenRes
+import androidx.annotation.StyleRes
+import androidx.constraintlayout.widget.ConstraintLayout
+import coil.load
+import com.infomaniak.lib.core.utils.getAttributes
+import com.infomaniak.lib.core.utils.setMarginsRelative
+import com.infomaniak.mail.R
+import com.infomaniak.mail.data.models.Attachment
+import com.infomaniak.mail.databinding.ViewAttachmentDetailsBinding
+import com.infomaniak.lib.core.R as RCore
+
+class AttachmentDetailsView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding by lazy { ViewAttachmentDetailsBinding.inflate(LayoutInflater.from(context), this) }
+
+    init {
+        attrs?.getAttributes(context, R.styleable.AttachmentDetailsView) {
+            val displayStyle = DisplayStyle.values()[getInteger(R.styleable.AttachmentDetailsView_displayStyle, 0)]
+            val displayResources = displayStyle.getStyleResources()
+            val iconSize = context.resources.getDimension(displayResources.iconSize).toInt()
+            val marginSize = context.resources.getDimension(displayResources.marginSize).toInt()
+
+            with(binding) {
+                fileName.setTextAppearance(displayResources.fileNameStyle)
+                fileSize.setTextAppearance(displayResources.fileSizeStyle)
+                icon.apply {
+                    layoutParams.height = iconSize
+                    layoutParams.width = iconSize
+                    setMarginsRelative(start = marginSize, end = marginSize)
+                }
+            }
+        }
+    }
+
+    fun setDetails(attachment: Attachment) = with(binding) {
+        fileName.text = attachment.name
+        fileSize.text = Formatter.formatShortFileSize(context, attachment.size)
+        icon.load(attachment.getFileTypeFromMimeType().icon)
+    }
+
+    private enum class DisplayStyle {
+        CHIP {
+            override fun getStyleResources() = DisplayStyleResources(
+                iconSize = R.dimen.standardIconSize,
+                marginSize = RCore.dimen.marginStandardSmall,
+                fileNameStyle = R.style.BodySmall,
+                fileSizeStyle = R.style.Label_Secondary,
+            )
+        },
+        BOTTOMSHEET {
+            override fun getStyleResources() = DisplayStyleResources(
+                iconSize = R.dimen.largeIconSize,
+                marginSize = RCore.dimen.marginStandard,
+                fileNameStyle = R.style.Body,
+                fileSizeStyle = R.style.Body_Secondary,
+            )
+        };
+
+        abstract fun getStyleResources(): DisplayStyleResources
+    }
```

Comment:
Can't we go we specify the parameters inside the Enum rather than using a data class?